x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/Ordinal.java

```diff
@@ -120,7 +126,21 @@ public boolean afterOrAt(Ordinal other) {
 
     public Object[] toArray() {
         return tiebreaker != null ?
-            new Object[] { timestamp, tiebreaker, implicitTiebreaker } 
-            : new Object[] { timestamp, implicitTiebreaker };
+            new Object[] { timestampAsString(), tiebreaker, implicitTiebreaker }
+            : new Object[] { timestampAsString(), implicitTiebreaker };
+    }
+
+    private static String instantToMillisString(Instant instant) {
+        long nanos = instant.getNano();
+        long millisOfSecond = nanos / 1_000_000L;
+        return (instant.getEpochSecond() * 1000L + millisOfSecond) + "." + (nanos - millisOfSecond * 1_000_000);
+    }
+
+    private static int timestampCompare(Object ts1, Object ts2) {
+        return ts1 instanceof Instant
+            ? (ts2 instanceof Instant ? ((Instant) ts1).compareTo((Instant) ts2) :
+                ((Instant) ts1).compareTo(Instant.ofEpochMilli(((Number) ts2).longValue())))
+            : (ts2 instanceof Instant ? Instant.ofEpochMilli(((Number) ts1).longValue()).compareTo((Instant) ts2) :
+                Long.compare(((Number) ts1).longValue(), ((Number) ts2).longValue()));
```

Comment:
Isn't the ternary operator being a bit overused here? Wouldn't an `if/else` be more easy to follow visually?

Also, looking at the changes in this PR, having `instanceof Instant` logic separate as a utility method would make more sense to show the difference between a date-with-nanos and date-with-millis?

Reply:
This has been refactored.